CONTRIBUTING.md

```diff
@@ -94,6 +94,107 @@ cd your-main-test-code
 npm link @wdio/cli
 ```
 
+## TypeScript definitions
+
+WebdriverIO provides their own type definitions for project that use TypeScript. Given the large amount of commands it would make it unmaintainable to __not__ automate the process of generating these. However there are certain edge cases where manual work is required. You can find all files responsible for the generating the typings here: https://github.com/webdriverio/webdriverio/tree/master/scripts/type-generation. You can trigger the process by calling:
+
+```sh
+$ npm run generate:typings
+```
+
+This will run the scripts in the directory shown above and generate the typings various packages located in `/packages/*`. Each package defines its typings differently:
+
+- the `webdriver` package defines all commands and their respective types in the json file of the `@wdio/protocols` package
+- the `webdriverio` package has the types defined in the comments above every command in `/packages/webdriverio/src/commands/**/*`
+- the `@wdio/allure-reporter` has its interface defined in comments within the service (`/packages/wdio-allure-reporter/src/index.js`)
+- all other packages have their type definitions directly in the `*.d.ts` in the root directory of that package (e.g. `/packages/wdio-cucumber-framework/cucumber-framework.d.ts`)
+
+Whenever you change the types in these packages, make sure you re-generate the types with the command shown above. In order to ensure that packages where type definitions have changed are released we keep the generated type definition in source control.
+
+### Adding Types for webdriverIO
```

Comment:
should WebdriverIO start from uppercase here?

Reply:
👍